src/main/java/com/example/test/sorters/Sorter.java

```diff
@@ -0,0 +1,33 @@
+package com.example.test.sorters;
+
+import com.example.test.R;
+
+public interface Sorter {
+
+    int[] sort(int[] array);
+
+    enum SortingMethod { // binded with strings.xml for changes
+        BUBBLESORT {
+            @Override
+            public int getCaptionStringId() {
+                return R.string.bubble_sort;
+            }
+        },
+        QUICKSORT {
+            @Override
+            public int getCaptionStringId() {
+                return R.string.quicksort;
+            }
+        },
+        COUNTINGSORT {
+            @Override
+            public int getCaptionStringId() {
+                return R.string.counting_sort;
+            }
+        };
+
+        public int getCaptionStringId() {
+            return 0;
```

Comment:
minor: to make sure that child class overrides this method, throw new AssertionError("getCaptionStringId() should be overriden!");
```suggestion
            throw new AssertionError("getCaptionStringId() should be overriden!");
            return 0;
```